py/desitarget/lyazcat.py

```diff
@@ -0,0 +1,463 @@
+    """Add the MgII absorption line finder data to the input zcat array.
+    
+    Parameters
+    ----------
+    zcat : :class:'~numpy.array'
+        The structured array that was created by make_new_zcat()
+    coaddname : class:'str'
+        The name of the coadd file corresponding to the zbest file used
+        in make_new_zcat()
+        
+    Returns
+    -------
+    :class:'~numpy.array'
+        The zcat array with SQUEzE data included in the columns:
+        * Z_ABS
+        * Z_ABS_CONF
+    """
+    tmark('    MgII Absorption data not yet added.')
+
+    return zcat
+
+
+def zcomb_selector(zcat, proc_flag=False):
+    """Compare results from redrock, QuasarNP, SQUEzE, and MgII data.
+    
+    Parameters
+    ----------
+    zcat : :class:'~numpy.array'
+        The structured array that was created by make_new_zcat()
+    proc_flag : :class:'bool'
+        Turn on extra comparison procedure.
+        
+    Returns
+    -------
+    :class:'~numpy.array'
+        The zcat array with SQUEzE data included in the columns:
+        * Z_COMB
+        * Z_COMB_PROB
+    """
+    zcat['Z_COMB'][:] = zcat['Z']
+    zcat['Z_COMB_PROB'][:] = 0.95
+
+    return zcat
+
+
+def zcat_writer(outputdir, zcat, outputname, qn_flag=False, sq_flag=False, abs_flag=False):
+    """Writes the zcat structured array out as a FITS file.
+    
+    Parameters
+    ----------
+    outputdir : :class:'str'
+        The directory where the zcat file will be written.
+    zcat : :class:'~numpy.array'
+        stuff
+    outputname : :class:'str'
+        The filename of the zcat output file.
+    qn_flag : :class:'bool'
+        Flag if QuasarNP data (or not) was added to the zcat file.
+    sq_flag : :class:'bool'
+        Flag if SQUEzE data (or not) was added to the zcat file.
+    abs_flag : :class:'bool'
+        Flag if MgII Absorption data (or not) was added to the zcat file.
+        
+    Returns
+    -------
+    :class:'str'
+        The filename, with path, of the FITS file written out.
+    """
+    tmark('    Creating file...')
+    full_outputname = os.path.join(outputdir, outputname)
+    
+    prim_hdu = fits.PrimaryHDU()
+    prim_hdu.header['QN_ADDED'] = str(qn_flag)
+    prim_hdu.header['SQ_ADDED'] = str(sq_flag)
+    prim_hdu.header['AB_ADDED'] = str(abs_flag)
+    
+    data_hdu = fits.BinTableHDU.from_columns(zcat, name='ZCATALOG')
+    data_out = fits.HDUList([prim_hdu, data_hdu])
+    
+    data_out.writeto(full_outputname)
+
+    return full_outputname
+
+
+def create_zcat(tile, night, petal_num,
+                zcatdir='/global/cfs/cdirs/desi/spectro/redux/daily', 
+                outputdir='/global/cfs/cdirs/desi/spectro/redux/daily',
+                qn_flag=False, sq_flag=False, abs_flag=False,
+                qnp_model='/global/cfs/cdirs/desi/science/lya/qn_models/boss_dr12/qn_train_coadd_indtrain_0_0_boss10.h5',
+                squeze_model='/global/homes/e/ebie/local/SQUEzE/data/BOSS_train_64plates_model.json',
```

Comment:
Use paths that can be shared by anyone at NERSC, not in your local home directory. I recommend making new spaces in `/global/cfs/cdirs/desi/target/catalogs/lya/qn_models` and `/global/cfs/cdirs/desi/target/catalogs/lya/sq_models`.

Reply:
Perfect. Exactly what I was looking for. I copied them over and changed the filename calls.